lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs

```diff
@@ -1255,47 +1257,60 @@ spec = do
             , expectErrorMessage (errMsg404NoWallet wid)
             ]
 
-    it "SHELLEY_TX_REDEEM_01 - Can redeem rewards from self" $ \ctx -> do
+    -- Reward wallets have 100k ADA UTxO and 1M ADA rewards,
+    -- fixtureWalletsWith have 100k ADA UTxO too
+    -- Verifying withdrawals for some boundary values
+    let rewardsMatrix = [ ("redeem sending exactly rewards", oneMillionAda)
+                        , ("redeem sending exactly UTxO", 100 * oneThousandAda)
+                        , ("redeem sending more than UTxO", 100 * oneThousandAda + oneAda)
```

Comment:
Isn't this case redundant with the first one :grimacing:  ?

Reply:
It depends on how you look at it :) 
 - First is sending amt = rewards when utxo is < rewards
 - This one is send amt = utxo + 1 when utxo is < rewards

They are different "edge cases" for me... (but you know implementation details better ofc)
So if you tell me we can safely drop one we can do it.

Reply:
addressed in https://github.com/input-output-hk/cardano-wallet/pull/1987/commits/ff01561b383b3e0989700fb54a2ff5ae0b12a4da